File: typescript/src/bebop_handlers/newLevel.ts
import GameState from '../classes/gameState';
import { NewLevelMsg } from '../models/pactheman.models';
import BebopHandler from './bebopHandler';

export default class NewLevelMsgHandler implements BebopHandler {
    handleMessage(sender: any, message: any): void {
        message = NewLevelMsg.decode(message);
        GameState.Instance.setResetCount();
        GameState.Instance.signalNewLevel(message);
        console.log('Starting new level');
    }
}

File: dart/lib/bebop_handlers/new_level.dart
import 'package:lean_dart_client/classes/game_state.dart';

import 'bebop_handler.dart';
import 'package:lean_dart_client/schemas.g.dart';

class NewLevelMsgHandler extends BebopHandler {
  @override
  void handleMessage(sender, message) {
    message = NewLevelMsg.decode(message);
    GameState.instance.setResetCounter();
    GameState.instance.signalNewLevel(message);
    print('Starting new level');
  }
}
